hotel/src/hil/rng.rs

```diff
@@ -0,0 +1,25 @@
+/// A Random Number Generator, based on the Rng trait from the rand crate.
+pub trait Rng {
+    /// Returns 4 bytes of random data.  Types that implement the Rng trait
+    /// only need to implement this function.
+    fn next_u32(&mut self) -> u32;
```

Comment:
I initially considered an asynchronous interface that used allow and subscribe for generating data but I figured it was probably unnecessarily complex.  In the hotel case the TRNG keeps a buffer of 2048 bytes of random data, which fills up very quickly, so it won't block unless the requested buffer is larger than that.

If you'd prefer the asynchronous interface I can re-spin this patch to use that.